### PR TITLE
Tracking doc: rewrite `@wdio/cli` package into TypeScript

### PR DESCRIPTION
**Pull request by christian-bromann**
This is a tracking issue for organising the effort of re-writing the code base from vanilla JS into TypeScript. If you are reading this and are interested getting involved, this is a great chance to do so. Here are the rules:

- if you are interested converting source files from `/packages/wdio-cli` into TypeScript please comment on this thread and mention the files you want to convert to avoid duplicate work
- there is no reason to convert the whole package at once - smaller PRs make it easier to review the code
- if you make a PR, reference this issue so we can keep track of the progress
- if you make a PR, label it with `TypeScript` and the package name `wdio-cli` so we can easier find all PRs
- if you make a PR, assign it to the `TypeScript Rewrite` project

Happy rewriting 😅

### CODE REVIEW

File: packages/wdio-cli/src/commands/config.ts [hunk 1]
```diff
@@ -27,31 +29,34 @@ export const cmdArgs = {
...
     const answers = await getAnswers(yes)
 
     const packageAnswers = ['reporters', 'runner', 'services', 'framework']
 
-    Object.keys(answers).forEach((key) => {
+    Object.keys(answers).forEach((key: keyof Questionnair) => {
         if (packageAnswers.includes(key)) {
             if (Array.isArray(answers[key])) {
-                answers[key] = answers[key].map(answer => convertPackageHashToObject(answer))
+                // @ts-ignore
```
Comment (mgrybyk) [inline packages/wdio-cli/src/commands/config.ts:50]: is it possible to remove ts ignore?

File: packages/wdio-cli/src/interface.ts [hunk 2]
```diff
@@ -101,8 +144,12 @@ export default class WDIOCLInterface extends EventEmitter {
         return _logger(...details)
     }
 
-    onTestError(payload) {
-        let error = { type: 'Error', message: typeof payload.error === 'string' ? payload.error : 'Unknown error.', stack: null }
+    onTestError (payload: CLIInterfaceEvent) {
+        const error: TestError = {
```
Comment (mgrybyk) [inline packages/wdio-cli/src/interface.ts:148]: `stack` is missing here

File: packages/wdio-cli/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import { BACKEND_CHOICES, MODE_OPTIONS, REGION_OPTION, COMPILER_OPTION_ANSWERS } from './constants'
```
Comment (mgrybyk) [inline packages/wdio-cli/src/types.ts:1]: shouldn't the extension of this file be `.d.ts`?

File: packages/wdio-cli/src/utils.ts [hunk 4]
```diff
@@ -135,21 +147,21 @@ export function findInConfig(config, type) {
     return config.match(regex)
 }
 
-export function replaceConfig(config, type, name) {
+export function replaceConfig (config: string, type: string, name: string) {
+    if (type === 'framework') {
```
Comment (mgrybyk) [inline packages/wdio-cli/src/utils.ts:151]: is it ok that the condition order changed?

**Review by mgrybyk — CHANGES_REQUESTED** [review]

File: packages/wdio-cli/src/types.ts (hunk 3 above)
Comment (christian-bromann) [reply packages/wdio-cli/src/types.ts:1]: There is no reason why it should (or are there any advantages I am missing?). Before the v7 release I will have all types properly be exposed from the `index.d.ts` file.

File: packages/wdio-cli/src/utils.ts (hunk 4 above)
Comment (christian-bromann) [reply packages/wdio-cli/src/utils.ts:151]: that should be irrelevant

File: packages/wdio-cli/src/types.ts (hunk 3 above)
Comment (mgrybyk) [reply packages/wdio-cli/src/types.ts:1]: Not quite sure, I supposed `.d.ts` extension is used for type definitions.

**Review by mgrybyk — APPROVED** [review]

**Pull request merged**